engine/src/main/java/org/terasology/rendering/dag/RenderTaskSystem.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag;
+
+import com.google.api.client.repackaged.com.google.common.base.Preconditions;
+import com.google.common.collect.Iterators;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+import org.lwjgl.opengl.Display;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.registry.Share;
+import static org.terasology.rendering.opengl.OpenGLUtils.bindDisplay;
+import static org.terasology.rendering.opengl.OpenGLUtils.renderFullscreenQuad;
+
+/**
+ * TODO: Add javadocs
+ */
+@RegisterSystem
+@Share(RenderTaskSystem.class)
```

Comment:
I think I find the creation of an entire system for this task not recommendable and the interaction it has with the WorldRenderer quite confusing. 

- I think the DebugControlSystem should inform the DebugOverlay (if enabled) that F11 has been pressed. 
- The DebugOverlay could then be an observable, the WorldRenderer being the observer.
- The DebugOverlay could then relay to the WorldRenderer the number of times F11 has been pressed.
- WorldRenderer then uses that number to process only a part of the task list.

This way you don't need to add a new BaseComponentSystem implementation.